src/compiler/checker.ts

```diff
@@ -16854,8 +16854,8 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     }
                 }
                 const propType = getTypeOfSymbol(prop);
-                return accessExpression && getAssignmentTargetKind(accessExpression) !== AssignmentKind.Definite ?
-                    getFlowTypeOfReference(accessExpression, propType) :
+                return accessExpression && getAssignmentTargetKind(accessExpression) !== AssignmentKind.Definite ? getFlowTypeOfReference(accessExpression, propType) :
+                    accessNode && isIndexedAccessTypeNode(accessNode) ? mapType(propType, type => type === missingType ? undefinedType : type) :

```

Comment:
```suggestion
                    accessNode && isIndexedAccessTypeNode(accessNode) && containsMissingType(propType) ? getUnionType([propType, undefinedType]) :

```
should save a lot of unneeded union traversals.

Reply:
thanks for the tip - I didn't realize that `getUnionType` might deduplicate `missingType` and `undefinedType` like this